kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProxyReconciler.java

```diff
@@ -122,9 +123,13 @@ public UpdateControl<KafkaProxy> reconcile(KafkaProxy primary,
     public ErrorStatusUpdateControl<KafkaProxy> updateErrorStatus(KafkaProxy proxy,
                                                                   Context<KafkaProxy> context,
                                                                   Exception e) {
+        if (e instanceof StaleReferentStatusException || e instanceof OperatorException && e.getCause() instanceof StaleReferentStatusException) {
+            LOGGER.info("Completed reconciliation of {}/{} with transient error {}", namespace(proxy), name(proxy), e.toString());
+            return ErrorStatusUpdateControl.noStatusUpdate();
+        }
         var uc = ErrorStatusUpdateControl.patchStatus(statusFactory.newUnknownConditionStatusPatch(proxy, Condition.Type.Ready, e));
         if (LOGGER.isInfoEnabled()) {
-            LOGGER.info("Completed reconciliation of {}/{} with error {}", namespace(proxy), name(proxy), e.toString());
+            LOGGER.info("Completed reconciliation of {}/{} with unhandled error {}", namespace(proxy), name(proxy), e.toString());
```

Comment:
I think all the reconcilers have the same logging, so I guess we should change them all?

Reply:
reverted this change after rewording the other case